.github/workflows/pr.yml

```diff
@@ -1,12 +1,12 @@
 name: Build CI
 
 on:
-  pull_request:
-    branches: [ "master" ]
+  pull_request_review:
+    types: [ submitted ]
 
 jobs:
   build:
-
+    if: github.event.review.state == 'approved'
```

Comment:
So build happens only when approve is set?
Do I understand this correctly?

If so, why exactly this way? If I commit some non-working code, I should be able to see that it fails on pipelines before somebody approves it and it becomes ready to be merged

Reply:
Or it happens twice?

Reply:
we do not do some magic staff in pr workflow.
Everyone should run tests before creating PR locally. CI just show its done and for to protect master branch allows reviewer from doing tests by himself. This workflow shouldnt run by every pushed commit because it obviously can have errors when fixing at WIP. And run when its ready for review.

Made change for run tests trigger added on pull request opened. To show reviewer checks pass before reviewing.

Reply:
> I did it on purpose.
Got it